### DISCUSSION TITLE
figure does not appear on website

### DISCUSSION THREAD

**Discussion by robpetrosino** [quarto-dev/quarto-cli#9166] [Q&A] [answered]
### Description

I finally managed to push my manuscript on GH pages, and everything is looking good except for the figures, which instead are not showing:  

<img width="806" alt="Screenshot 2024-03-22 at 3 49 20 PM" src="https://github.com/quarto-dev/quarto-cli/assets/30778296/efe9f45b-d4ad-49b1-b8cb-748a4ebf6f03">

The two figures are two `ggplot` objects (e.g., [here](https://github.com/robpetrosino/frequency-effects_masked-priming/blob/b4680c0f90db91d5d3fc435deaf828cdd31181ef/index.qmd#L236)) that do compile in the pdf version, but not in the html one. 

The link to the GH page is: [https://robpetrosino.github.io/frequency-effects_masked-priming/#sec-intro](https://robpetrosino.github.io/frequency-effects_masked-priming)
The link to the repo: [https://github.com/robpetrosino/frequency-effects_masked-priming/tree/main](https://github.com/robpetrosino/frequency-effects_masked-priming/tree/main
)
Does anyone know how to solve the issue?

**Comment by mcanouil** [comment]
As said your "example" is quite too big.
You need to go step by step to remove everything not related to the issue to get something small.
Then and only then, you'll be able to figure out by yourself the issue, or at least we would be able to help.


I am sorry but we don't have the time to debug thousands lines document.

Also, as said in the other thread you need to fix your code cells and use only one syntax, YAML is the recommended one in Quarto.
But anyway, in both Rmarkdown/Quarto, mixing syntaxes is wrong and can lead to unintended side effects.

**Reply by robpetrosino** [reply]
Fair, sorry! I am trying to figure what is going wrong.

**Comment by cderv** [comment]
How to do you publish to gh-pages exactly ? 

Your gh-pages branch is missing the `index_files/` folder where the image is. Locally when doing `quarto render index.qmd --to html`  I see the folder in the `_manuscript` output directory.

**Reply by robpetrosino** [reply]
Hello, thank you for replying. Not sure what you mean by this. Can you expand on it? I followed the instructions on the website, so I just `git push` to the origin...

**Reply by cderv** [reply]
I meant how did you publish : https://robpetrosino.github.io/frequency-effects_masked-priming/#sec-intro

So the rendered output, not the source file in repo.

I see no `index_files` in `gh-pages` branch; if you did push it manually maybe you are missing it ? 
https://github.com/robpetrosino/frequency-effects_masked-priming/tree/gh-pages

Or did you use `quarto publish gh-pages`

**Reply by robpetrosino** [reply]
I see what you are saying. The `gh-pages` branch in the local repo does have the `index_files` folder, but it is empty. 

I am not sure what I did wrong. I just imported the template, worked on the qmd files, and pushed it to the remote repo as indicated in the guide. I definitely did not use the `quarto publish gh-pages` command. 

What is also weird is that the `index_files` folder in the `main` branch still contains the images of the template (even though all files of the template were deleted).

**Reply by robpetrosino** [reply]
I ran into this other [guide](https://quarto.org/docs/publishing/github-pages.html#publishing) from the Quarto website, and tried to `quarto publish gh-pages`. Now the `gh-pages` branch has the `index_files`, and the figures show up on the website!

So, for the future the right flow to publish is:
1. commit everything on the main branch
2. push on the main branch to origin
3. `quarto publish gh-pages`

Right?

**Reply by mcanouil** [reply]
> I am not sure what I did wrong. I just imported the template, worked on the qmd files, and pushed it to the remote repo as indicated in the guide. I definitely did not use the quarto publish gh-pages command.

Sorry, I am confused. What guide exactly are you referring to?

**Reply by robpetrosino** [reply]
This is the first guide I was following: [https://quarto.org/docs/manuscripts/publishing.html](https://quarto.org/docs/manuscripts/publishing.html), which assumes you clone the quarto template. Earlier this morning I found this other [guide](https://quarto.org/docs/publishing/github-pages.html#publishing) that instead tells you what to do in case you didn't clone the template and/or didn't close all branches from the template. 

I followed the first guide, but only after executing the flow 1-3 above the `index_files` folder appeared on the `gh-pages` branch, and the figures show up on the website!

**Reply by mcanouil** [reply, marked as answer]
I see, the section is not accurate.
- https://quarto.org/docs/manuscripts/publishing.html#push-to-publish

1. `quarto render` is needed to update the `_freeze` directory with computation results (not very clear currently and could be improved)
2. `quarto publish gh-pages` is needed before the GitHub Action can take over the publishing part.

**Reply by robpetrosino** [reply]
The page you linked here says to "commit and push the changes to Github", which I interpreted as push in the main branch. But now you seem to say that it is not strictly necessary for the GH Page to get updated, and we actually need to `quarto publish gh-pages` after `quarto render` (rather than just push in the main/origin branch). Is that correct?

**Reply by mcanouil** [reply]
> commit and push the changes to Github

"changes" refers to your source code (Quarto documents, `_quarto.yml`, etc.) AND `_freeze` directory.

`quarto publish gh-pages` is only needed once, the very first time you publish to GitHub Pages (it initialise almost everything on GitHub side).

The part just before <https://quarto.org/docs/publishing/github-pages.html#ignoring-output> is outdated and not needed anymore. `quarto publish gh-pages` does that for you.
<img width="877" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/157e88e9-3423-43fb-b196-57639a4ea8ef">

**Reply by robpetrosino** [reply]
Got it. So just to be clear -- for future updates of the same page, I will just have to push in the main branch, and the new version should be deployed in the `gh-pages` branch too. Correct?

**Reply by mcanouil** [reply]
Yes, using the GitHub Actions workflow you should have if you cloned/forked the template repository.
